### ISSUE TITLE
no types

### ISSUE THREAD

**Issue by Johnrobmiller** [benwinding/quill-image-compress#30]
There are no ts types for this library.

**Comment by benwinding** [comment]
This is true. It's currently a JS lib, would you like me to add types?

**Comment by Johnrobmiller** [comment]
🎉

**Comment by chpio** [comment]
This is not fixed. Would need a `"types": "types/index.d.ts",` line in the package.json.

The usage seem to have changed as well (breaking change! and needs an update in the readme):

```ts
import ReactQuill, {Quill as OriginalQuill} from 'react-quill';
import QuillImageCompress from 'quill-image-compress';

OriginalQuill.register('modules/imageCompressor', QuillImageCompress);

const MODULES: Record<string, unknown> = {
  imageCompressor: {
    quality: 0.7,
    maxWidth: 1000,
    maxHeight: 1000,
    imageType: 'image/jpeg',
  },
};
```

(*imageCompressor* instead of *imageCompress*)

is #27 related to this change?

**Comment by benwinding** [comment]
> This is not fixed. Would need a `"types": "types/index.d.ts"`, line in the package.json.

Good catch! Thanks for reporting! 🙏 

> The usage seem to have changed as well (breaking change! and needs an update in the readme):

This isn't a breaking change at all, as the `Quill.register()` command is what signals the options object key.

For example, this here works too:

``` js
OriginalQuill.register('modules/imageCompressor123', QuillImageCompress);
const MODULES: Record<string, unknown> = {
  imageCompressor123: {
    ....
  },
};
```
Will change existing examples to all be `imageCompress` for clarity though 👍

**Comment by chpio** [comment]
yeah, you're right. My mistake. How does quill know what property it does need to use as the constructor?

**Comment by benwinding** [comment]
> How does quill know what property it does need to use as the constructor?

Quill internally assigns the key to the class provided, using `Quill.register('module/key', MyClass)` so it doesn't matter what the class is called, it's referenced by the `key`. (_also in JS classes can be passed around like functions_)

This specific example is also using the "default export" of the JS module. Which can be named anything, as it's not a "named export"

Which means this will work too!

``` js
import MyQuillImageCompress123 from 'quill-image-compress';

OriginalQuill.register('modules/imageCompressor123', MyQuillImageCompress123);

const MODULES: Record<string, unknown> = {
  imageCompressor123: {
    ....
  },
};
```

JavaScript is a fun language! :rage1:

**Comment by chpio** [comment]
> This specific example is also using the "default export" of the JS module.

yeah, but you do not do that

https://github.com/benwinding/quill-image-compress/blob/7f1ebda29594934c8b0579a5cd6d4d4101728019/src/index.ts#L4-L5

And i somehow misinterpreted

```ts
import QuillImageCompress from 'quill-image-compress';
```

for

```ts
import * as QuillImageCompress from 'quill-image-compress';
```

**Comment by chpio** [comment]
see my edit from the last post. that implies that my `types` line was wrong, and should have been `"types": "types/quill.imageCompressor.d.ts"`

**Comment by benwinding** [comment]
Thanks again!! 😅 
Yeah I just needed to export the default module in the index, fixed here 4bfbefe
Should be deployed as version `1.2.29`
